OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ExternalExportController.cs

```diff
@@ -65,4 +67,52 @@ public async Task<IActionResult> GetWorkshopsByFilter([FromQuery] DateTime updat
             return StatusCode(500, $"An error occurred: {ex.Message}");
         }
     }
+    
+    /// <summary>
+    /// Get Directions that match filter's parameters.
+    /// </summary>
+    /// <param name="offsetFilter">Filter to get a part of all directions that were found.</param>
+    /// <returns><see cref="SearchResult{DirectionInfoDto}"/>, or no content.</returns>
+    [HttpGet]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(SearchResult<DirectionInfoDto>))]
+    [ProducesResponseType(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    [Route("directions")]
+    public async Task<IActionResult> GetDirectionsByFilter([FromQuery] OffsetFilter offsetFilter)
+    {
+        try
```

Comment:
Just an idea. This 
```try { var result = await DoSomething(); return mapped(result); } catch { return 500 code; }```
seems like a pattern that could be refactored via `Task<T>` extension method `ProtectAndMap` that awaits passed taks, and either catches and transforms errors to StatusCode or transforms result using passed map function, like this
```= > await DoSomething().ProtectAndMap(mapped);```